dartagnan/src/main/java/com/dat3m/dartagnan/wmm/analysis/RelationAnalysis.java

```diff
@@ -863,6 +900,61 @@ public Knowledge visitSameAddress(Relation rel) {
             }
             return new Knowledge(may, enableMustSets ? must : EMPTY_SET);
         }
+
+        @Override
+        public Knowledge visitSameScope(Relation rel, ArrayList<String> scopes) {
+            Set<Tuple> must = new HashSet<>();
+            List<Load> loadEvents = program.getEvents(Load.class);
+            List<Store> storeEvents = program.getEvents(Store.class);
+            List<Fence> fenceEvents = program.getEvents(Fence.class);
```

Comment:
variables are used only once, no need to introduce the variables

Reply:
Accepted